### PR TITLE
fixes #8385 - save last taxonomy on logout

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

File: app/controllers/concerns/foreman/controller/users_mixin.rb [hunk 1]
```diff
@@ -57,13 +57,24 @@ def update_sub_hostgroups_owners
     sub_hg.each { |hg| hg.users << @user }
   end
 
+  def save_current_taxonomies(user, options = {})
+    session ||= options.fetch(:session, {})
+    ['location', 'organization'].each do |taxonomy|
+      if (current_taxonomy = session["{taxonomy}_id"])
```
Comment (dLobatog) [inline app/controllers/concerns/foreman/controller/users_mixin.rb]: Is the if/else condition necessary here? It seems like

``` ruby
user.update_attribute("last_#{taxonomy}", session["#{taxonomy}_id"])
```

would suffice as `session['randomvalue']` returns nil anyway. 

Notice you have a typo in the string interpolation to get `#{taxonomy}_id`

**Comment by dLobatog** [comment]
@stbenjam This looks good and works fine, please address the concern I mentioned in the comment and :+1: to merge. 

I fixed the typo myself to test it out and it works well, it should fix the currently broken tests in Jenkins too.

**Comment by dLobatog** [comment]
@stbenjam We're close to merge this, it's been a while since I tested it but most likely it's still working. Mind to fix the test & ping me to review & merge?

**Comment by domcleal** [comment]
A test remains broken on this PR.

**Comment by tbrisker** [comment]
@stbenjam,
Thank you very much for your efforts! 
This PR has been open for a very long time with no activity. 
Please indicate if you plan to continue working on it or not; if not, feel free to close it. 
If you do not respond either way within a couple of weeks, I will assume this PR is abandoned and will go ahead and close it for you.

**Comment by tbrisker** [comment]
Dear @stbenjam ,
It has been 20 days since my last comment with no replies. 
I am assuming this pull request has been abandoned and closing it.
If you wish to continue working on this in the future, feel free to reopen.
On behalf of the entire Foreman team, I wish to thank you for your efforts here and hope to see more contributions from you in the future.

**Pull request closed without merging**